segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/TableExtensionConfig.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.tables;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import io.pravega.segmentstore.contracts.Attributes;
+import io.pravega.segmentstore.contracts.SegmentProperties;
+import io.pravega.segmentstore.contracts.tables.TableAttributes;
+import java.time.Duration;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeoutException;
+import lombok.Builder;
+import lombok.Data;
+
+/**
+ * Configuration for {@link ContainerTableExtensionImpl} and sub-components.
+ */
+@Data
+@Builder
+class TableExtensionConfig {
+    /**
+     * The maximum unindexed length ({@link SegmentProperties#getLength() - {@link TableAttributes#INDEX_OFFSET}}) of a
+     * Segment for which {@link ContainerKeyIndex} {@code triggerCacheTailIndex} can be invoked.
+     */
+    @Builder.Default
+    private int maxTailCachePreIndexLength = EntrySerializer.MAX_BATCH_SIZE * 4;
```

Comment:
did not understand why multiplied by 4

Reply:
MaxBatchSize == 32MB (32 * MaxEntrySize).
I wrote 128MB as a function of MaxBatchSize, and implicitly as a function of MaxEntrySize.